app/web_framework/cluster_base_handler.py

```diff
@@ -46,6 +47,11 @@ def _handle_request_exception(self, ex):
         self.set_status(status_code)
         self.finish()
 
+    def on_finish(self):
+        http_request_duration_seconds.labels(self.request.method,  # pylint: disable=no-member
+                                             self.route,  # pylint: disable=no-member
```

Comment:
`self.route` is not a thing, is it? I imagine this would be failing for every request.